File: mang5o/220927/pg-12924.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12924
# 숫자의 표현

def solution(n):
    acc_sum = [0]  # 누적합 행렬
    for idx in range(n):
        acc_sum.append(acc_sum[-1] + idx + 1)

    start_ptr = 0
    end_ptr = 0
    answer = 0
    while end_ptr <= n:
        if acc_sum[end_ptr] - acc_sum[start_ptr] == n:
            # print("{} ~ {}".format(start_ptr + 1, end_ptr))
            answer += 1
            end_ptr += 1
        elif acc_sum[end_ptr] - acc_sum[start_ptr] < n:
            end_ptr += 1
        else:
            start_ptr += 1

    return answer


print(solution(15))

File: hwjeon/220926/pg12924.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
	int num = (n + 1) / 2;

	int sum = 0, answer = 1, j = 1;



	for (int i = 1; i <= num; i++) {

		sum += i;

		if (sum >= n) {

			while (sum > n) sum -= j++;

			if (sum == n) answer++;

		}

	}

	return n != 1 ? answer : 1;
}